virtcontainers/qemu_arch_base.go

```diff
@@ -461,70 +475,80 @@ func networkModelToQemuType(model NetInterworkingModel) govmmQemu.NetDeviceType
 	}
 }
 
-func (q *qemuArchBase) appendNetwork(devices []govmmQemu.Device, endpoint Endpoint) []govmmQemu.Device {
+func genericNetwork(endpoint Endpoint, vhost, nestedRun bool, index int) govmmQemu.NetDevice {
```

Comment:
I wonder if wouldn't be better to return a pointer and in the caller check `if d == nil {`  or return an `error` like `genericImage`, what do you think ?

Reply:
I think that depends if we want to `log` an `error` or not. I would prefer returning an `error` and `logging it`

Reply:
I changed `genericNetwork` and `genericBlockDevice` to return an `error` on failure.